### PR TITLE
feat: Accessibility Support for Secure Guest Link - WPB - 673

### PR DESCRIPTION
**Pull request by agisilaos**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

With this PR I introduce Accessibility Support for Secure Guest Link Feature. A few noticeable changes are:

- Introduced a Scroll View so everything is scrollable and accessible when using Dynamic Type
- Introduced accessibility labels hints and in some cases announcements for almost all components. 

The announcements are specifically for announcing if there's anything wrong regarding the password validation via VoiceOver. 

Last but not least, I cleaned up the code a bit. 

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by KaterinaWire — APPROVED** [review]

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation Options/CreateSecureGuestLink/CreateSecureGuestLinkViewController.swift [hunk 1]
```diff
@@ -252,10 +318,29 @@ class CreateSecureGuestLinkViewController: UIViewController, CreatePasswordSecur
...
+            UIAccessibility.post(
+                notification: .announcement,
+                argument: SecureGuestLinkAccessibilityLocale.SecuredGuestLinkPasswordTextfield.announcement
+            )
+        } else {
+            UIAccessibility.post(
+                notification: .announcement,
+                argument: SecureGuestLinkAccessibilityLocale.SecuredGuestLinkPasswordValidatedTextField.announcement
+            )
+        }
```
Comment (netbe) [inline wire-ios/Wire-iOS/Sources/UserInterface/Conversation Options/CreateSecureGuestLink/CreateSecureGuestLinkViewController.swift]: readability improvement + if there's a way to make this `SecureGuestLinkAccessibilityLocale.SecuredGuestLinkPasswordTextfield.announcement` shorter :)
 
```suggestion
    func announcePasswordValidationErrorForVoiceOver(for textField: ValidatedTextField) {
       let argument =  textField == securedGuestLinkPasswordTextfield ? SecureGuestLinkAccessibilityLocale.SecuredGuestLinkPasswordTextfield.announcement : SecureGuestLinkAccessibilityLocale.SecuredGuestLinkPasswordValidatedTextField.announcement
        
            UIAccessibility.post(
                notification: .announcement,
                argument: argument
            )
        }
```

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation Options/CreateSecureGuestLink/CreateSecureGuestLinkViewController.swift [hunk 2]
```diff
@@ -27,13 +27,25 @@ class CreateSecureGuestLinkViewController: UIViewController, CreatePasswordSecur
     typealias ViewColors = SemanticColors.View
     typealias LabelColors = SemanticColors.Label
     typealias SecuredGuestLinkWithPasswordLocale = L10n.Localizable.SecuredGuestLinkWithPassword
+    typealias SecureGuestLinkAccessibilityLocale = L10n.Accessibility.CreateSecureGuestLink
```
Comment (netbe) [inline wire-ios/Wire-iOS/Sources/UserInterface/Conversation Options/CreateSecureGuestLink/CreateSecureGuestLinkViewController.swift:30]:
```suggestion
    typealias accessibilityLocale = L10n.Accessibility.CreateSecureGuestLink
```

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation Options/CreateSecureGuestLink/CreateSecureGuestLinkViewController.swift [hunk 3]
```diff
@@ -27,13 +27,25 @@ class CreateSecureGuestLinkViewController: UIViewController, CreatePasswordSecur
     typealias ViewColors = SemanticColors.View
     typealias LabelColors = SemanticColors.Label
     typealias SecuredGuestLinkWithPasswordLocale = L10n.Localizable.SecuredGuestLinkWithPassword
```
Comment (netbe) [inline wire-ios/Wire-iOS/Sources/UserInterface/Conversation Options/CreateSecureGuestLink/CreateSecureGuestLinkViewController.swift:29]:
```suggestion
    typealias passwordLocale = L10n.Localizable.SecuredGuestLinkWithPassword
```

**Review by netbe — APPROVED** [review]
LGTM:)

**Pull request merged**